tests/unit/s2n_self_talk_psk_test.c

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+
+/* Include source to call static function s2n_match_psk_identity */
+#include "tls/extensions/s2n_client_psk.c"
+
+#define TEST_PSK_WIRE_INDEX 1
+#define TEST_PSK_HMAC S2N_HMAC_SHA256
+
+uint8_t test_shared_identity[] = "test shared identity";
+uint8_t test_shared_secret[] = "test shared secret";
+
+static s2n_result setup_client_psks(struct s2n_connection *client_conn)
+{
+    ENSURE_REF(client_conn);
+
+    /* Setup other client PSK */
+    uint8_t other_client_data[] = "other client data";
+    struct s2n_psk *other_client_psk = NULL;
+    GUARD_RESULT(s2n_array_pushback(&client_conn->psk_params.psk_list, (void**) &other_client_psk));
+    GUARD_AS_RESULT(s2n_psk_init(other_client_psk, S2N_PSK_TYPE_EXTERNAL));
+    GUARD_AS_RESULT(s2n_psk_new_identity(other_client_psk, other_client_data, sizeof(other_client_data)));
+    GUARD_AS_RESULT(s2n_psk_new_secret(other_client_psk, other_client_data, sizeof(other_client_data)));
+    other_client_psk->hmac_alg = S2N_HMAC_SHA256;
```

Comment:
TEST_PSK_HMAC

Reply:
Actually we wish to keep this different from the shared_psk's hmac (i.e TEST_PSK_HMAC). Will change it  `S2N_HMAC_SHA384` to be more specific.